### PR TITLE
[vue] let dependabot manage some vue dependencies

### PR DESCRIPTION
**Pull request by atomfrede**
This PR follows https://github.com/jhipster/generator-jhipster/pull/12972

It uses husky and lint-staged from constants, updates some vue dependencies where possible and save (e.g. terser plugin 5.x is not compatible with webpack 4). Some of the common dependencies are can now be managed by dependabot. Didn't put vue core on purpose yet until we are on vue3.

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/generator-constants.js [hunk 1]
```diff
@@ -48,8 +48,8 @@ const KAFKA_VERSION = '5.5.2';
 const JACKSON_DATABIND_NULLABLE_VERSION = '0.2.1';
 
 // NPM packages version
-const HUSKY_VERSION = '4.2.5';
-const LINT_STAGED_VERSION = '10.4.2';
+const HUSKY_VERSION = '4.3.0';
```
Comment (murdos) [inline generators/generator-constants.js:51]: These versions should probably be managed by https://github.com/jhipster/generator-jhipster/blob/main/generators/client/templates/common/package.json now

**Comment by atomfrede** [comment]
`husky` and `lint-staged` are now also manged by common package.json

File: generators/client/templates/vue/package.json.ejs [hunk 2]
```diff
@@ -96,18 +96,16 @@ limitations under the License.
...
-    "husky": "4.3.0",
+    "husky": "VERSION_MANAGED_BY_CLIENT_COMMON",
+    "lint-staged": "VERSION_MANAGED_BY_CLIENT_COMMON,
     <%_ } _%>
     "jest": "26.4.2",
     "jest-junit": "11.1.0",
     "jest-serializer-vue": "2.0.2",
     "jest-sonar-reporter": "2.0.0",
     "jest-vue-preprocessor": "1.7.1",
-    <%_ if (!skipCommitHook) { _%>
```
Comment (murdos) [inline generators/client/templates/vue/package.json.ejs:108]: I think we should keep this condition

Comment (atomfrede) [reply generators/client/templates/vue/package.json.ejs:108]: It is still there. Just put both deps under one condition:

```
    <%_ if (!skipCommitHook) { _%>
    "husky": "VERSION_MANAGED_BY_CLIENT_COMMON",
    "lint-staged": "VERSION_MANAGED_BY_CLIENT_COMMON,
    <%_ } _%>
```

File: generators/generator-constants.js [hunk 3]
```diff
@@ -47,9 +47,6 @@ const JACOCO_VERSION = '0.8.6';
 const KAFKA_VERSION = '5.5.2';
 const JACKSON_DATABIND_NULLABLE_VERSION = '0.2.1';
 
-// NPM packages version
-const HUSKY_VERSION = '4.2.5';
```
Comment (pascalgrimaud) [inline generators/generator-constants.js:51]: this is needed by `package.json` in server/ folder, that's why the build failed I think

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

File: generators/generator-constants.js (hunk 3 above)
Comment (atomfrede) [reply generators/generator-constants.js:51]: Thanks for the hint, totally right.

File: generators/client/templates/angular/package.json.ejs [hunk 4]
```diff
@@ -79,8 +79,8 @@ limitations under the License.
     "@types/sockjs-client": "1.1.1",
 <%_ } _%>
 <%_ if (!skipCommitHook) { _%>
-    "husky": "<%= HUSKY_VERSION %>",
-    "lint-staged": "<%= LINT_STAGED_VERSION %>",
+    "husky": "VERSION_MANAGED_BY_CLIENT_COMMON",
+    "lint-staged": "VERSION_MANAGED_BY_CLIENT_COMMON,
```
Comment (mshima) [inline generators/client/templates/angular/package.json.ejs:83]:
```suggestion
    "lint-staged": "VERSION_MANAGED_BY_CLIENT_COMMON",
```
Same for others.

Comment (atomfrede) [reply generators/client/templates/angular/package.json.ejs:83]: :facepalm:

**Review by pascalgrimaud — APPROVED** [review]

**Pull request merged**